packages/starlight/index.ts

```diff
@@ -1,3 +1,18 @@
+/**
+ * These triple-slash directives defines dependencies to various declaration files that will be
+ * loaded when a user imports the Starlight integration in their Astro configuration file. These
+ * directives must be first at the top of the file and can only be preceded by this comment.
+ *
+ * - `./locals.d.ts` registers the various Astro middlewares used by Starlight and also defines a
+ * 	 namespace reserved for Starlight (only used for i18n at the moment) that can be extended by
+ *   plugins using module augmentation and interface merging.
+ * - `./i18n.d.ts` imports the original `i18next` types and extends to configure the Starlight
+ *   namespace (note that this can only be done in a different file due to a top-level `import`
+ *   making it non-ambient).
```

Comment:
My initial thought was to have everything in this file to avoid showing these comments to a user when they "Go to Definition", e.g. for `locals.t`, but now seeing the comments with your changes, I think it's fine and I made it more of an issue than it actually is. I'll update the PR tomorrow morning :thumbsup:

Reply:
Ah yeah didn’t consider these files as user-facing. But still, I agree, I think these comments are OK also there (plus not sure quite how often people will click through). If it turns out to be more frequent, we could even consider something more tailored to that audience.